arches/app/models/resource.py

```diff
@@ -669,15 +670,23 @@ def get_relations(resourceinstanceid, start, limit, resourceinstance_graphid=Non
         restricted_instances = get_restricted_instances(user, se) if user is not None else []
         for relation in resource_relations["relations"]:
             relation = model_to_dict(relation)
-            try:
-                preflabel = get_preflabel_from_valueid(relation["relationshiptype"], lang)
-                relation["relationshiptype_label"] = preflabel["value"] or ""
-            except:
-                relation["relationshiptype_label"] = relation["relationshiptype"] or ""
-
             resourceid_to = relation["resourceinstanceidto"]
             resourceid_from = relation["resourceinstanceidfrom"]
-            if resourceid_to not in restricted_instances and resourceid_from not in restricted_instances:
+            resourceinstanceto_graphid = relation["resourceinstanceto_graphid"]
+            resourceinstancefrom_graphid = relation["resourceinstancefrom_graphid"]
+
+            if (
+                resourceid_to not in restricted_instances 
+                and resourceid_from not in restricted_instances
+                and user_can_read_graph(user, resourceinstanceto_graphid)
+                and user_can_read_graph(user, resourceinstancefrom_graphid)
```

Comment:
Maybe rather than creating a new function 'user_can_read_graph' that calls ```user_has_resource_model_permissions```, we could call ```user_has_resource_model_permissions(user, ["models.read_nodegroup"], graph_id=resourceinstancefrom_graph_id)``` directly here?